### PR TITLE
drop cards v2

### LINKED ISSUE #9797: Drop card v2
iterating fast, lets play more on the social factor 
- after observations that price and status are not so relevant, here is updated card

image

updated in figma handoff drop page: https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=394-6126&mode=design&t=UwOjVaVtynz3fLJf-11

## Minting live
- the minted/supply indicates its status, if minted = supply, its obvious its minted out 
- if its 0/supply, it could be scheduled or live drop that has no mints yet
   - for that there is the right part (collected by part) 

image

- for state when its live but no one minted yet

when someone mints, it starts showing bubbles (collected by) 

- bubbles will have disabled hover for now, and let's see how it works 

## Scheduled
image

- shows minted/supply and also the date 
- when the time to drop < 24 hours, it starts to show the hours - like in drop page etc.

## Ended
- minted = supply 
- collected by bubbles

## Other
- one major change is that the preview img should have fixed aspect ratio of 16:9
- the bottom part scales accordingly

https://github.com/kodadot/nft-gallery/assets/90852205/08646c57-0533-4cce-998c-6b2fb2f39474

let me know if anything 

thanks!

### PR DESCRIPTION
**Pull request by iMac7**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #9797 
- [ ] Requires deployment <worker/stick/speck>

#### Before submitting pull request, please make sure:

- [ ] My contribution builds **clean without any errors or warnings**
- [ ] I've merged recent default branch -- **main** and I've no conflicts
- [ ] I've tried to respect high code quality standards
- [ ] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ahp/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer?target=14dLrAvjDw7cFLaxM7VWR6YtHd7bGZY2QpzXDhAJJ4yv6oGw&usdamount=undefined)

#### Community participation

- [x] [Are you at Koda Ecosystem Telegram?](https://t.me/koda_eco)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screenshot_20240320_235722

### CODE REVIEW

**Comment by exezbcz** [comment]
could you please check figma once again 

image


image

height should be definitely lower

File: components/collection/drop/CollectedByAvatar.vue [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-      <Avatar :size="38" :value="address" />
+      <Avatar :size="isDropsPage ? 26 : 38" :value="address" />
     </template>
   </IdentityModuleIdentityPopover>
 </template>
 
 <script lang="ts" setup>
 const props = defineProps<{
   address: string
+  isDropsPage: boolean
```
Comment (Jarsen136) [inline components/collection/drop/CollectedByAvatar.vue:14]: I would use `size: 'small' | 'medium'` instead

File: components/collection/drop/CollectedBy.vue [hunk 2]
```diff
@@ -16,9 +18,11 @@
 <script setup lang="ts">
 const props = defineProps<{
   addresses: string[]
+  maxAddressCount?: number
+  isDropsPage?: boolean
```
Comment (Jarsen136) [inline components/collection/drop/CollectedBy.vue:22]: the same. We should avoid this kind of props `isXXXpage`

File: composables/collectionActivity/useCollectionActivity.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ export const useCollectionActivity = ({ collectionId }) => {
 
   const queryPrefix = queryPrefixMap[urlPrefix.value] || 'subsquid'
   const variables = computed(() => ({
-    id: collectionId.value,
+    id: collectionId.value || collectionId.collection,
```
Comment (Jarsen136) [inline composables/collectionActivity/useCollectionActivity.ts:18]: It does not make sense... What you should change is how it calls `useCollectionActivity`

check https://github.com/kodadot/nft-gallery/pull/9830/files#diff-329c47d069c9f4255f7e36138bebbe999e113c5fb3dbe0270dde55867341a578R74

File: components/collection/drop/CollectedBy.vue [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
+    class="rounded-full h-[62px] md:w-auto border border-k-shade inline-flex items-center justify-between px-3"
+    :class="{ 'h-full border-none': isDropsPage }">
     <div class="flex items-center">
       <CollectionDropCollectedByAvatar
         v-for="(address, index) in addresses.slice(0, maxAddressCount)"
         :key="address"
         :class="[zIndexMap[index], { '-ml-2': index > 0 }]"
-        :address="address" />
+        :address="address"
+        :is-drops-page="isDropsPage || false" />
```
Comment (Jarsen136) [inline components/collection/drop/CollectedBy.vue:11]: It's unnecessary to set `false` as the default value

**Comment by yangwao** [comment]
Great job @iMac7 
On mobile it needs some work tho

**Comment by yangwao** [comment]
Release is estimated at least 2pm UTC+1, I believe you'll manage 👌😁

File: components/collection/drop/CollectedBy.vue [hunk 5]
```diff
@@ -16,9 +18,11 @@
 <script setup lang="ts">
 const props = defineProps<{
   addresses: string[]
+  maxAddressCount?: number
+  size?: string
```
Comment (Jarsen136) [inline components/collection/drop/CollectedBy.vue:22]:
```suggestion
  size?: 'small' | 'medium'
```

File: components/collection/drop/CollectedByAvatar.vue [hunk 6]
```diff
@@ -1,14 +1,17 @@
...
-      <Avatar :size="38" :value="address" />
+      <Avatar :size="size === 'small' ? 26 : 38" :value="address" />
     </template>
   </IdentityModuleIdentityPopover>
 </template>
 
 <script lang="ts" setup>
 const props = defineProps<{
   address: string
+  size?: string
```
Comment (Jarsen136) [inline components/collection/drop/CollectedByAvatar.vue:14]: the same

File: components/drops/DropCard.vue [hunk 7]
```diff
@@ -11,29 +11,30 @@
...
             </div>
+            <TimeTag
+              v-if="drop.dropStartTime && !ownerAddresses.length"
+              :drop-start-time="drop.dropStartTime"
+              :drop-status="drop.status" />
+            <CollectionDropCollectedBy
+              v-if="ownerAddresses.length"
+              :addresses="ownerAddresses"
+              :max-address-count="3"
+              :size="'small'" />
```
Comment (Jarsen136) [inline components/drops/DropCard.vue:37]:
```suggestion
              size="small" />
```

**Comment by yangwao** [comment]
> what resolution/device were you using for that?

Fold5

**Comment by yangwao** [comment]
There are still these two bugs

File: components/collection/drop/CollectedByAvatar.vue [hunk 8]
```diff
@@ -1,14 +1,17 @@
...
+      <Avatar :size="size === 'small' ? 26 : 38" :value="address" />
     </template>
   </IdentityModuleIdentityPopover>
 </template>
 
 <script lang="ts" setup>
 const props = defineProps<{
   address: string
+  size?: 'small' | 'medium' | 'large'
 }>()
```
Comment (Jarsen136) [inline components/collection/drop/CollectedByAvatar.vue:15]: Why do you need `large` here?
```
const props = withDefaults(
  defineProps<{
    size?: 'small' | 'medium'
  }>(),
  {
    size: 'medium',
  },
)
```

**Review by Jarsen136 — COMMENTED** [review]
otherwise code lgtm

File: components/collection/drop/CollectedByAvatar.vue (hunk 8 above)
Comment (Jarsen136) [reply components/collection/drop/CollectedByAvatar.vue]: @iMac7 Please add the default value `size: 'medium',`  here

**Review by Jarsen136 — APPROVED** [review]

**Comment by Jarsen136** [comment]
image

- It does not work for me


image

- not align

**Comment by yangwao** [comment]
@iMac7 comments above plus you broke original one at drop page

image

**Comment by yangwao** [comment]
Plus, the estimation calculation seems wrong. 
If drop plans to be at 3 pm my local time and it is 12:31 it tells me it would be at 2pm which is wrong.

image

**Comment by yangwao** [comment]
@prury can you please keep close eye on this?

**Comment by exezbcz** [comment]
right/left padding is smaller than on desktop 
image

y-align center please
image

**Comment by yangwao** [comment]
> it's centered on my end. resolution?

you can check yourself what we see
https://deploy-preview-9830--koda-canary.netlify.app/

colors are back gud
image

File: components/identity/module/IdentityPopover.vue [hunk 9]
```diff
@@ -26,6 +26,10 @@ import { useIdentitySoldData } from '../utils/useIdentity'
 import IdentityPopoverHeader from './IdentityPopoverHeader.vue'
 import IdentityPopoverFooter from './IdentityPopoverFooter.vue'
 
+const props = defineProps<{
+  size?: 'small' | 'medium'
+}>()
+
```
Comment (Jarsen136) [inline components/identity/module/IdentityPopover.vue:32]: WHY? It seems not related to the size. 

If you want to control its shown, you should use another proper name.

**Review by Jarsen136 — CHANGES_REQUESTED** [review]

**Comment by Jarsen136** [comment]
> > > it's centered on my end. resolution?
> > 
> > 
> > you can check yourself what we see https://deploy-preview-9830--koda-canary.netlify.app/
> > colors are back gud image
> 
>  2024-03-21.15-34-08.mp4 
> somehow this change is not in deploy preview


Here is the reason. You could fix it in the code.
image

**Comment by yangwao** [comment]
> somehow this change is not in deploy preview

are you sure committed all changes?

**Comment by yangwao** [comment]
@iMac7 please resolve merge conflicts and I'll probably merge it

**Comment by Jarsen136** [comment]
> > somehow this change is not in deploy preview
> 
> cc @Jarsen136

I mean there is a bug in the code regarding the aligned issue. I have attached the screenshot above. It's not related to how it deploy : )

**Comment by yangwao** [comment]
> I mean there is a bug in the code regarding the aligned issue.

you can do it in follow-up if you want, apparently, @iMac7 can not figure it on his own :D

**Review by Jarsen136 — APPROVED** [review]

**Comment by yangwao** [comment]
> I will refactor on my end as well

cool you have less than five minutes otherwise, I'm closing this PR

**Comment by yangwao** [comment]
image

@Jarsen136 can you try to resolve merge conflicts from this PR? And y-alingment of course :D

**Comment by prury** [comment]
max amount is wrong, should be 128: 
image

date prob wrong: 
image
